From 4f5a9e4779a2dc6a26a35d406dcfd74ac52d24ba Mon Sep 17 00:00:00 2001
From: "sungho1.shin" <sungho1.shin@lge.com>
Date: Tue, 26 Jul 2022 14:41:54 +0900
Subject: [PATCH] Update enact version to 4.5

:Release Notes:
Update enact version to 4.5

:Detailed Notes:
This commit modifies the sample enact application provided
for webos to be compatiable with enact version 4.5.

:Testing Performed:
Verified that build is successfully done.

:Issues Addressed:
[WRO-9280] Support enact 4.5 upgrade
---
 .../npm-shrinkwrap.json                       | 831 ++++++++++++++----
 .../com.example.app.iotivity/package.json     |  16 +-
 .../com.example.app.iotivity/src/App/App.js   |   1 -
 .../com.example.app.iotivity/src/index.js     |  19 +-
 .../src/views/Client.js                       |  14 +-
 .../src/views/MainPanel.js                    |   9 +-
 .../src/views/Server.js                       |  18 +-
 7 files changed, 689 insertions(+), 219 deletions(-)

diff --git a/resource/csdk/stack/samples/webos/com.example.app.iotivity/npm-shrinkwrap.json b/resource/csdk/stack/samples/webos/com.example.app.iotivity/npm-shrinkwrap.json
index 097210db3..5025900a0 100644
--- a/resource/csdk/stack/samples/webos/com.example.app.iotivity/npm-shrinkwrap.json
+++ b/resource/csdk/stack/samples/webos/com.example.app.iotivity/npm-shrinkwrap.json
@@ -1,258 +1,729 @@
 {
   "name": "com.example.app.iotivity",
   "version": "1.0.0",
+  "lockfileVersion": 2,
+  "requires": true,
+  "packages": {
+    "": {
+      "name": "com.example.app.iotivity",
+      "version": "1.0.0",
+      "license": "LGE",
+      "dependencies": {
+        "@enact/core": "^4.5.0",
+        "@enact/i18n": "^4.5.0",
+        "@enact/moonstone": "^4.5.0",
+        "@enact/spotlight": "^4.5.0",
+        "@enact/ui": "^4.5.0",
+        "@enact/webos": "^4.5.0",
+        "prop-types": "^15.5.10",
+        "react": "^18.0.0",
+        "react-dom": "^18.0.0",
+        "react-redux": "^5.0.7",
+        "redux": "^3.7.2",
+        "fbjs": "^3.0.0",
+        "redux-thunk": "^2.2.0"
+      }
+    },
+    "node_modules/@babel/runtime": {
+      "version": "7.18.9",
+      "resolved": "https://registry.npmjs.org/@babel/runtime/-/runtime-7.18.9.tgz",
+      "integrity": "sha512-lkqXDcvlFT5rvEjiu6+QYO+1GXrEHRo2LOtS7E4GtX5ESIZOgepqsZBVIj6Pv+a6zqsya9VCgiK1KAK4BvJDAw==",
+      "dependencies": {
+        "regenerator-runtime": "^0.13.4"
+      },
+      "engines": {
+        "node": ">=6.9.0"
+      }
+    },
+    "node_modules/@enact/core": {
+      "version": "4.5.0",
+      "resolved": "https://registry.npmjs.org/@enact/core/-/core-4.5.0.tgz",
+      "integrity": "sha512-A8539ZuCetFBVUqFqYDVr3abUFI7U4sLKlSyQMcjdvTcdkNwX8AAB4PV4yY09mxVZpVkDGJxcGfTtxsd2Ixl1w==",
+      "dependencies": {
+        "classnames": "^2.3.1",
+        "invariant": "^2.2.4",
+        "prop-types": "^15.8.1",
+        "ramda": "^0.28.0",
+        "react": "^18.0.0",
+        "react-dom": "^18.0.0",
+        "react-is": "^18.0.0",
+        "warning": "^4.0.3"
+      }
+    },
+    "node_modules/@enact/i18n": {
+      "version": "4.5.0",
+      "resolved": "https://registry.npmjs.org/@enact/i18n/-/i18n-4.5.0.tgz",
+      "integrity": "sha512-W6vebdHPtMj31wwhR45VD9RIP+tJwPoWDRKgLTzRuoahcIpmGHV8+C3i1E+1pMXvweY0Z4P7rjyXhUu2tRZcow==",
+      "dependencies": {
+        "@enact/core": "^4.5.0",
+        "prop-types": "^15.8.1",
+        "ramda": "^0.28.0",
+        "react": "^18.0.0",
+        "react-dom": "^18.0.0",
+        "xhr": "^2.6.0"
+      },
+      "peerDependencies": {
+        "ilib": "^14.14.0 || ^14.14.0-webos1"
+      }
+    },
+    "node_modules/@enact/moonstone": {
+      "version": "4.5.0",
+      "resolved": "https://registry.npmjs.org/@enact/moonstone/-/moonstone-4.5.0.tgz",
+      "integrity": "sha512-ELjJ4NMCJy6JjRnRgEQYoyV9fs0yP8YiLTRxvXd3hDcqzYJ7NYIuDN6X1tT/9nh+ZSl+w5sIiaWMEgzHqJOiBw==",
+      "dependencies": {
+        "@enact/core": "^4.5.0-rc.1",
+        "@enact/i18n": "^4.5.0-rc.1",
+        "@enact/spotlight": "^4.5.0-rc.1",
+        "@enact/ui": "^4.5.0-rc.1",
+        "classnames": "^2.3.1",
+        "ilib": "^14.14.0",
+        "invariant": "^2.2.4",
+        "prop-types": "^15.8.1",
+        "ramda": "^0.28.0",
+        "react": "^18.0.0",
+        "react-dom": "^18.0.0",
+        "warning": "^4.0.3"
+      },
+      "peerDependencies": {
+        "ilib": "^14.14.0 || ^14.14.0-webos1"
+      }
+    },
+    "node_modules/@enact/spotlight": {
+      "version": "4.5.0",
+      "resolved": "https://registry.npmjs.org/@enact/spotlight/-/spotlight-4.5.0.tgz",
+      "integrity": "sha512-LUAc0Al5aSBHWgS7njyIdvcWrmLjOKiR+zXCKs8lLRd+pHD0b+uf29vDUBG8mMwzwW1YpMwmyXjr+Oi2QiOv5w==",
+      "dependencies": {
+        "@enact/core": "^4.5.0",
+        "classnames": "^2.3.1",
+        "prop-types": "^15.8.1",
+        "ramda": "^0.28.0",
+        "react": "^18.0.0",
+        "react-dom": "^18.0.0",
+        "warning": "^4.0.3"
+      }
+    },
+    "node_modules/@enact/ui": {
+      "version": "4.5.0",
+      "resolved": "https://registry.npmjs.org/@enact/ui/-/ui-4.5.0.tgz",
+      "integrity": "sha512-0kTQFe5arYai1um6WXFLY6S2ATwkXB9Qw2vrsS8oNX/oDm6T+KAUQ9UlbJCsus/6l6uF/RCFV+tO+52Lmldbdg==",
+      "dependencies": {
+        "@enact/core": "^4.5.0",
+        "classnames": "^2.3.1",
+        "direction": "^1.0.4",
+        "invariant": "^2.2.4",
+        "prop-types": "^15.8.1",
+        "ramda": "^0.28.0",
+        "react": "^18.0.0",
+        "react-dom": "^18.0.0",
+        "react-is": "^18.0.0",
+        "warning": "^4.0.3"
+      }
+    },
+    "node_modules/@enact/webos": {
+      "version": "4.5.0",
+      "resolved": "https://registry.npmjs.org/@enact/webos/-/webos-4.5.0.tgz",
+      "integrity": "sha512-XO0GQm/DfbRGkLfvkO/iSF2gTN7e5vh9wt3bs18hSBB3FLkPg1WP4Fvg9dIsbR2wrqu2inxVRgKV46TTH1M+bw==",
+      "dependencies": {
+        "@enact/core": "^4.5.0",
+        "prop-types": "^15.8.1",
+        "react": "^18.0.0",
+        "react-dom": "^18.0.0"
+      }
+    },
+    "node_modules/classnames": {
+      "version": "2.3.1",
+      "resolved": "https://registry.npmjs.org/classnames/-/classnames-2.3.1.tgz",
+      "integrity": "sha512-OlQdbZ7gLfGarSqxesMesDa5uz7KFbID8Kpq/SxIoNGDqY8lSYs0D+hhtBXhcdB3rcbXArFr7vlHheLk1voeNA=="
+    },
+    "node_modules/direction": {
+      "version": "1.0.4",
+      "resolved": "https://registry.npmjs.org/direction/-/direction-1.0.4.tgz",
+      "integrity": "sha512-GYqKi1aH7PJXxdhTeZBFrg8vUBeKXi+cNprXsC1kpJcbcVnV9wBsrOu1cQEdG0WeQwlfHiy3XvnKfIrJ2R0NzQ==",
+      "bin": {
+        "direction": "cli.js"
+      },
+      "funding": {
+        "type": "github",
+        "url": "https://github.com/sponsors/wooorm"
+      }
+    },
+    "node_modules/dom-walk": {
+      "version": "0.1.2",
+      "resolved": "https://registry.npmjs.org/dom-walk/-/dom-walk-0.1.2.tgz",
+      "integrity": "sha512-6QvTW9mrGeIegrFXdtQi9pk7O/nSK6lSdXW2eqUspN5LWD7UTji2Fqw5V2YLjBpHEoU9Xl/eUWNpDeZvoyOv2w=="
+    },
+    "node_modules/global": {
+      "version": "4.4.0",
+      "resolved": "https://registry.npmjs.org/global/-/global-4.4.0.tgz",
+      "integrity": "sha512-wv/LAoHdRE3BeTGz53FAamhGlPLhlssK45usmGFThIi4XqnBmjKQ16u+RNbP7WvigRZDxUsM0J3gcQ5yicaL0w==",
+      "dependencies": {
+        "min-document": "^2.19.0",
+        "process": "^0.11.10"
+      }
+    },
+    "node_modules/hoist-non-react-statics": {
+      "version": "3.3.2",
+      "resolved": "https://registry.npmjs.org/hoist-non-react-statics/-/hoist-non-react-statics-3.3.2.tgz",
+      "integrity": "sha512-/gGivxi8JPKWNm/W0jSmzcMPpfpPLc3dY/6GxhX2hQ9iGj3aDfklV4ET7NjKpSinLpJ5vafa9iiGIEZg10SfBw==",
+      "dependencies": {
+        "react-is": "^16.7.0"
+      }
+    },
+    "node_modules/hoist-non-react-statics/node_modules/react-is": {
+      "version": "16.13.1",
+      "resolved": "https://registry.npmjs.org/react-is/-/react-is-16.13.1.tgz",
+      "integrity": "sha512-24e6ynE2H+OKt4kqsOvNd8kBpV65zoxbA4BVsEOB3ARVWQki/DHzaUoC5KuON/BiccDaCCTZBuOcfZs70kR8bQ=="
+    },
+    "node_modules/ilib": {
+      "version": "14.14.0",
+      "resolved": "https://registry.npmjs.org/ilib/-/ilib-14.14.0.tgz",
+      "integrity": "sha512-QuhH6qwJYq8c368+px4KA3KXfWfRC8XDt/zGNdwOz5C19ZOkPaxIh9LRr1RmzHEPJsd0mKj/JGV3pnzx1xXljg=="
+    },
+    "node_modules/invariant": {
+      "version": "2.2.4",
+      "resolved": "https://registry.npmjs.org/invariant/-/invariant-2.2.4.tgz",
+      "integrity": "sha512-phJfQVBuaJM5raOpJjSfkiD6BpbCE4Ns//LaXl6wGYtUBY83nWS6Rf9tXm2e8VaK60JEjYldbPif/A2B1C2gNA==",
+      "dependencies": {
+        "loose-envify": "^1.0.0"
+      }
+    },
+    "node_modules/is-function": {
+      "version": "1.0.2",
+      "resolved": "https://registry.npmjs.org/is-function/-/is-function-1.0.2.tgz",
+      "integrity": "sha512-lw7DUp0aWXYg+CBCN+JKkcE0Q2RayZnSvnZBlwgxHBQhqt5pZNVy4Ri7H9GmmXkdu7LUthszM+Tor1u/2iBcpQ=="
+    },
+    "node_modules/js-tokens": {
+      "version": "4.0.0",
+      "resolved": "https://registry.npmjs.org/js-tokens/-/js-tokens-4.0.0.tgz",
+      "integrity": "sha512-RdJUflcE3cUzKiMqQgsCu06FPu9UdIJO0beYbPhHN4k6apgJtifcoCtT9bcxOpYBtpD2kCM6Sbzg4CausW/PKQ=="
+    },
+    "node_modules/lodash": {
+      "version": "4.17.21",
+      "resolved": "https://registry.npmjs.org/lodash/-/lodash-4.17.21.tgz",
+      "integrity": "sha512-v2kDEe57lecTulaDIuNTPy3Ry4gLGJ6Z1O3vE1krgXZNrsQ+LFTGHVxVjcXPs17LhbZVGedAJv8XZ1tvj5FvSg=="
+    },
+    "node_modules/lodash-es": {
+      "version": "4.17.21",
+      "resolved": "https://registry.npmjs.org/lodash-es/-/lodash-es-4.17.21.tgz",
+      "integrity": "sha512-mKnC+QJ9pWVzv+C4/U3rRsHapFfHvQFoFB92e52xeyGMcX6/OlIl78je1u8vePzYZSkkogMPJ2yjxxsb89cxyw=="
+    },
+    "node_modules/loose-envify": {
+      "version": "1.4.0",
+      "resolved": "https://registry.npmjs.org/loose-envify/-/loose-envify-1.4.0.tgz",
+      "integrity": "sha512-lyuxPGr/Wfhrlem2CL/UcnUc1zcqKAImBDzukY7Y5F/yQiNdko6+fRLevlw1HgMySw7f611UIY408EtxRSoK3Q==",
+      "dependencies": {
+        "js-tokens": "^3.0.0 || ^4.0.0"
+      },
+      "bin": {
+        "loose-envify": "cli.js"
+      }
+    },
+    "node_modules/min-document": {
+      "version": "2.19.0",
+      "resolved": "https://registry.npmjs.org/min-document/-/min-document-2.19.0.tgz",
+      "integrity": "sha512-9Wy1B3m3f66bPPmU5hdA4DR4PB2OfDU/+GS3yAB7IQozE3tqXaVv2zOjgla7MEGSRv95+ILmOuvhLkOK6wJtCQ==",
+      "dependencies": {
+        "dom-walk": "^0.1.0"
+      }
+    },
+    "node_modules/object-assign": {
+      "version": "4.1.1",
+      "resolved": "https://registry.npmjs.org/object-assign/-/object-assign-4.1.1.tgz",
+      "integrity": "sha512-rJgTQnkUnH1sFw8yT6VSU3zD3sWmu6sZhIseY8VX+GRu3P6F7Fu+JNDoXfklElbLJSnc3FUQHVe4cU5hj+BcUg==",
+      "engines": {
+        "node": ">=0.10.0"
+      }
+    },
+    "node_modules/parse-headers": {
+      "version": "2.0.5",
+      "resolved": "https://registry.npmjs.org/parse-headers/-/parse-headers-2.0.5.tgz",
+      "integrity": "sha512-ft3iAoLOB/MlwbNXgzy43SWGP6sQki2jQvAyBg/zDFAgr9bfNWZIUj42Kw2eJIl8kEi4PbgE6U1Zau/HwI75HA=="
+    },
+    "node_modules/process": {
+      "version": "0.11.10",
+      "resolved": "https://registry.npmjs.org/process/-/process-0.11.10.tgz",
+      "integrity": "sha512-cdGef/drWFoydD1JsMzuFf8100nZl+GT+yacc2bEced5f9Rjk4z+WtFUTBu9PhOi9j/jfmBPu0mMEY4wIdAF8A==",
+      "engines": {
+        "node": ">= 0.6.0"
+      }
+    },
+    "node_modules/prop-types": {
+      "version": "15.8.1",
+      "resolved": "https://registry.npmjs.org/prop-types/-/prop-types-15.8.1.tgz",
+      "integrity": "sha512-oj87CgZICdulUohogVAR7AjlC0327U4el4L6eAvOqCeudMDVU0NThNaV+b9Df4dXgSP1gXMTnPdhfe/2qDH5cg==",
+      "dependencies": {
+        "loose-envify": "^1.4.0",
+        "object-assign": "^4.1.1",
+        "react-is": "^16.13.1"
+      }
+    },
+    "node_modules/prop-types/node_modules/react-is": {
+      "version": "16.13.1",
+      "resolved": "https://registry.npmjs.org/react-is/-/react-is-16.13.1.tgz",
+      "integrity": "sha512-24e6ynE2H+OKt4kqsOvNd8kBpV65zoxbA4BVsEOB3ARVWQki/DHzaUoC5KuON/BiccDaCCTZBuOcfZs70kR8bQ=="
+    },
+    "node_modules/ramda": {
+      "version": "0.28.0",
+      "resolved": "https://registry.npmjs.org/ramda/-/ramda-0.28.0.tgz",
+      "integrity": "sha512-9QnLuG/kPVgWvMQ4aODhsBUFKOUmnbUnsSXACv+NCQZcHbeb+v8Lodp8OVxtRULN1/xOyYLLaL6npE6dMq5QTA==",
+      "funding": {
+        "type": "opencollective",
+        "url": "https://opencollective.com/ramda"
+      }
+    },
+    "node_modules/react": {
+      "version": "18.2.0",
+      "resolved": "https://registry.npmjs.org/react/-/react-18.2.0.tgz",
+      "integrity": "sha512-/3IjMdb2L9QbBdWiW5e3P2/npwMBaU9mHCSCUzNln0ZCYbcfTsGbTJrU/kGemdH2IWmB2ioZ+zkxtmq6g09fGQ==",
+      "dependencies": {
+        "loose-envify": "^1.1.0"
+      },
+      "engines": {
+        "node": ">=0.10.0"
+      }
+    },
+    "node_modules/react-dom": {
+      "version": "18.2.0",
+      "resolved": "https://registry.npmjs.org/react-dom/-/react-dom-18.2.0.tgz",
+      "integrity": "sha512-6IMTriUmvsjHUjNtEDudZfuDQUoWXVxKHhlEGSk81n4YFS+r/Kl99wXiwlVXtPBtJenozv2P+hxDsw9eA7Xo6g==",
+      "dependencies": {
+        "loose-envify": "^1.1.0",
+        "scheduler": "^0.23.0"
+      },
+      "peerDependencies": {
+        "react": "^18.2.0"
+      }
+    },
+    "node_modules/react-is": {
+      "version": "18.2.0",
+      "resolved": "https://registry.npmjs.org/react-is/-/react-is-18.2.0.tgz",
+      "integrity": "sha512-xWGDIW6x921xtzPkhiULtthJHoJvBbF3q26fzloPCK0hsvxtPVelvftw3zjbHWSkR2km9Z+4uxbDDK/6Zw9B8w=="
+    },
+    "node_modules/react-lifecycles-compat": {
+      "version": "3.0.4",
+      "resolved": "https://registry.npmjs.org/react-lifecycles-compat/-/react-lifecycles-compat-3.0.4.tgz",
+      "integrity": "sha512-fBASbA6LnOU9dOU2eW7aQ8xmYBSXUIWr+UmF9b1efZBazGNO+rcXT/icdKnYm2pTwcRylVUYwW7H1PHfLekVzA=="
+    },
+    "node_modules/react-redux": {
+      "version": "5.1.2",
+      "resolved": "https://registry.npmjs.org/react-redux/-/react-redux-5.1.2.tgz",
+      "integrity": "sha512-Ns1G0XXc8hDyH/OcBHOxNgQx9ayH3SPxBnFCOidGKSle8pKihysQw2rG/PmciUQRoclhVBO8HMhiRmGXnDja9Q==",
+      "dependencies": {
+        "@babel/runtime": "^7.1.2",
+        "hoist-non-react-statics": "^3.3.0",
+        "invariant": "^2.2.4",
+        "loose-envify": "^1.1.0",
+        "prop-types": "^15.6.1",
+        "react-is": "^16.6.0",
+        "react-lifecycles-compat": "^3.0.0"
+      },
+      "peerDependencies": {
+        "react": "^0.14.0 || ^15.0.0-0 || ^16.0.0-0",
+        "redux": "^2.0.0 || ^3.0.0 || ^4.0.0-0"
+      }
+    },
+    "node_modules/react-redux/node_modules/react-is": {
+      "version": "16.13.1",
+      "resolved": "https://registry.npmjs.org/react-is/-/react-is-16.13.1.tgz",
+      "integrity": "sha512-24e6ynE2H+OKt4kqsOvNd8kBpV65zoxbA4BVsEOB3ARVWQki/DHzaUoC5KuON/BiccDaCCTZBuOcfZs70kR8bQ=="
+    },
+    "node_modules/redux": {
+      "version": "3.7.2",
+      "resolved": "https://registry.npmjs.org/redux/-/redux-3.7.2.tgz",
+      "integrity": "sha512-pNqnf9q1hI5HHZRBkj3bAngGZW/JMCmexDlOxw4XagXY2o1327nHH54LoTjiPJ0gizoqPDRqWyX/00g0hD6w+A==",
+      "dependencies": {
+        "lodash": "^4.2.1",
+        "lodash-es": "^4.2.1",
+        "loose-envify": "^1.1.0",
+        "symbol-observable": "^1.0.3"
+      }
+    },
+    "node_modules/redux-thunk": {
+      "version": "2.4.1",
+      "resolved": "https://registry.npmjs.org/redux-thunk/-/redux-thunk-2.4.1.tgz",
+      "integrity": "sha512-OOYGNY5Jy2TWvTL1KgAlVy6dcx3siPJ1wTq741EPyUKfn6W6nChdICjZwCd0p8AZBs5kWpZlbkXW2nE/zjUa+Q==",
+      "peerDependencies": {
+        "redux": "^4"
+      }
+    },
+    "node_modules/regenerator-runtime": {
+      "version": "0.13.9",
+      "resolved": "https://registry.npmjs.org/regenerator-runtime/-/regenerator-runtime-0.13.9.tgz",
+      "integrity": "sha512-p3VT+cOEgxFsRRA9X4lkI1E+k2/CtnKtU4gcxyaCUreilL/vqI6CdZ3wxVUx3UOUg+gnUOQQcRI7BmSI656MYA=="
+    },
+    "node_modules/scheduler": {
+      "version": "0.23.0",
+      "resolved": "https://registry.npmjs.org/scheduler/-/scheduler-0.23.0.tgz",
+      "integrity": "sha512-CtuThmgHNg7zIZWAXi3AsyIzA3n4xx7aNyjwC2VJldO2LMVDhFK+63xGqq6CsJH4rTAt6/M+N4GhZiDYPx9eUw==",
+      "dependencies": {
+        "loose-envify": "^1.1.0"
+      }
+    },
+    "node_modules/symbol-observable": {
+      "version": "1.2.0",
+      "resolved": "https://registry.npmjs.org/symbol-observable/-/symbol-observable-1.2.0.tgz",
+      "integrity": "sha512-e900nM8RRtGhlV36KGEU9k65K3mPb1WV70OdjfxlG2EAuM1noi/E/BaW/uMhL7bPEssK8QV57vN3esixjUvcXQ==",
+      "engines": {
+        "node": ">=0.10.0"
+      }
+    },
+    "node_modules/warning": {
+      "version": "4.0.3",
+      "resolved": "https://registry.npmjs.org/warning/-/warning-4.0.3.tgz",
+      "integrity": "sha512-rpJyN222KWIvHJ/F53XSZv0Zl/accqHR8et1kpaMTD/fLCRxtV8iX8czMzY7sVZupTI3zcUTg8eycS2kNF9l6w==",
+      "dependencies": {
+        "loose-envify": "^1.0.0"
+      }
+    },
+    "node_modules/xhr": {
+      "version": "2.6.0",
+      "resolved": "https://registry.npmjs.org/xhr/-/xhr-2.6.0.tgz",
+      "integrity": "sha512-/eCGLb5rxjx5e3mF1A7s+pLlR6CGyqWN91fv1JgER5mVWg1MZmlhBvy9kjcsOdRk8RrIujotWyJamfyrp+WIcA==",
+      "dependencies": {
+        "global": "~4.4.0",
+        "is-function": "^1.0.1",
+        "parse-headers": "^2.0.0",
+        "xtend": "^4.0.0"
+      }
+    },
+    "node_modules/xtend": {
+      "version": "4.0.2",
+      "resolved": "https://registry.npmjs.org/xtend/-/xtend-4.0.2.tgz",
+      "integrity": "sha512-LKYU1iAXJXUgAXn9URjiu+MWhyUXHsvfp7mcuYm9dSUKK0/CjtrUwFAxD82/mCWbtLsGjFIad0wIsod4zrTAEQ==",
+      "engines": {
+        "node": ">=0.4"
+      }
+    }
+  },
   "dependencies": {
+    "@babel/runtime": {
+      "version": "7.18.9",
+      "resolved": "https://registry.npmjs.org/@babel/runtime/-/runtime-7.18.9.tgz",
+      "integrity": "sha512-lkqXDcvlFT5rvEjiu6+QYO+1GXrEHRo2LOtS7E4GtX5ESIZOgepqsZBVIj6Pv+a6zqsya9VCgiK1KAK4BvJDAw==",
+      "requires": {
+        "regenerator-runtime": "^0.13.4"
+      }
+    },
     "@enact/core": {
-      "version": "1.15.0",
-      "from": "@enact/core@^1.8.0",
-      "resolved": "https://registry.npmjs.org/@enact/core/-/core-1.15.0.tgz"
+      "version": "4.5.0",
+      "resolved": "https://registry.npmjs.org/@enact/core/-/core-4.5.0.tgz",
+      "integrity": "sha512-A8539ZuCetFBVUqFqYDVr3abUFI7U4sLKlSyQMcjdvTcdkNwX8AAB4PV4yY09mxVZpVkDGJxcGfTtxsd2Ixl1w==",
+      "requires": {
+        "classnames": "^2.3.1",
+        "invariant": "^2.2.4",
+        "prop-types": "^15.8.1",
+        "ramda": "^0.28.0",
+        "react": "^18.0.0",
+        "react-dom": "^18.0.0",
+        "react-is": "^18.0.0",
+        "warning": "^4.0.3"
+      }
     },
     "@enact/i18n": {
-      "version": "1.15.0",
-      "from": "@enact/i18n@^1.8.0",
-      "resolved": "https://registry.npmjs.org/@enact/i18n/-/i18n-1.15.0.tgz"
+      "version": "4.5.0",
+      "resolved": "https://registry.npmjs.org/@enact/i18n/-/i18n-4.5.0.tgz",
+      "integrity": "sha512-W6vebdHPtMj31wwhR45VD9RIP+tJwPoWDRKgLTzRuoahcIpmGHV8+C3i1E+1pMXvweY0Z4P7rjyXhUu2tRZcow==",
+      "requires": {
+        "@enact/core": "^4.5.0",
+        "prop-types": "^15.8.1",
+        "ramda": "^0.28.0",
+        "react": "^18.0.0",
+        "react-dom": "^18.0.0",
+        "xhr": "^2.6.0"
+      }
     },
     "@enact/moonstone": {
-      "version": "1.15.0",
-      "from": "@enact/moonstone@^1.8.0",
-      "resolved": "https://registry.npmjs.org/@enact/moonstone/-/moonstone-1.15.0.tgz"
+      "version": "4.5.0",
+      "resolved": "https://registry.npmjs.org/@enact/moonstone/-/moonstone-4.5.0.tgz",
+      "integrity": "sha512-ELjJ4NMCJy6JjRnRgEQYoyV9fs0yP8YiLTRxvXd3hDcqzYJ7NYIuDN6X1tT/9nh+ZSl+w5sIiaWMEgzHqJOiBw==",
+      "requires": {
+        "@enact/core": "^4.5.0-rc.1",
+        "@enact/i18n": "^4.5.0-rc.1",
+        "@enact/spotlight": "^4.5.0-rc.1",
+        "@enact/ui": "^4.5.0-rc.1",
+        "classnames": "^2.3.1",
+        "ilib": "^14.14.0",
+        "invariant": "^2.2.4",
+        "prop-types": "^15.8.1",
+        "ramda": "^0.28.0",
+        "react": "^18.0.0",
+        "react-dom": "^18.0.0",
+        "warning": "^4.0.3"
+      }
     },
     "@enact/spotlight": {
-      "version": "1.15.0",
-      "from": "@enact/spotlight@^1.8.0",
-      "resolved": "https://registry.npmjs.org/@enact/spotlight/-/spotlight-1.15.0.tgz"
+      "version": "4.5.0",
+      "resolved": "https://registry.npmjs.org/@enact/spotlight/-/spotlight-4.5.0.tgz",
+      "integrity": "sha512-LUAc0Al5aSBHWgS7njyIdvcWrmLjOKiR+zXCKs8lLRd+pHD0b+uf29vDUBG8mMwzwW1YpMwmyXjr+Oi2QiOv5w==",
+      "requires": {
+        "@enact/core": "^4.5.0",
+        "classnames": "^2.3.1",
+        "prop-types": "^15.8.1",
+        "ramda": "^0.28.0",
+        "react": "^18.0.0",
+        "react-dom": "^18.0.0",
+        "warning": "^4.0.3"
+      }
     },
     "@enact/ui": {
-      "version": "1.15.0",
-      "from": "@enact/ui@^1.8.0",
-      "resolved": "https://registry.npmjs.org/@enact/ui/-/ui-1.15.0.tgz"
+      "version": "4.5.0",
+      "resolved": "https://registry.npmjs.org/@enact/ui/-/ui-4.5.0.tgz",
+      "integrity": "sha512-0kTQFe5arYai1um6WXFLY6S2ATwkXB9Qw2vrsS8oNX/oDm6T+KAUQ9UlbJCsus/6l6uF/RCFV+tO+52Lmldbdg==",
+      "requires": {
+        "@enact/core": "^4.5.0",
+        "classnames": "^2.3.1",
+        "direction": "^1.0.4",
+        "invariant": "^2.2.4",
+        "prop-types": "^15.8.1",
+        "ramda": "^0.28.0",
+        "react": "^18.0.0",
+        "react-dom": "^18.0.0",
+        "react-is": "^18.0.0",
+        "warning": "^4.0.3"
+      }
     },
     "@enact/webos": {
-      "version": "1.15.0",
-      "from": "@enact/webos@^1.8.0",
-      "resolved": "https://registry.npmjs.org/@enact/webos/-/webos-1.15.0.tgz"
-    },
-    "asap": {
-      "version": "2.0.6",
-      "from": "asap@~2.0.3",
-      "resolved": "https://registry.npmjs.org/asap/-/asap-2.0.6.tgz"
-    },
-    "change-emitter": {
-      "version": "0.1.6",
-      "from": "change-emitter@^0.1.2",
-      "resolved": "https://registry.npmjs.org/change-emitter/-/change-emitter-0.1.6.tgz"
+      "version": "4.5.0",
+      "resolved": "https://registry.npmjs.org/@enact/webos/-/webos-4.5.0.tgz",
+      "integrity": "sha512-XO0GQm/DfbRGkLfvkO/iSF2gTN7e5vh9wt3bs18hSBB3FLkPg1WP4Fvg9dIsbR2wrqu2inxVRgKV46TTH1M+bw==",
+      "requires": {
+        "@enact/core": "^4.5.0",
+        "prop-types": "^15.8.1",
+        "react": "^18.0.0",
+        "react-dom": "^18.0.0"
+      }
     },
     "classnames": {
-      "version": "2.2.5",
-      "from": "classnames@~2.2.5",
-      "resolved": "https://registry.npmjs.org/classnames/-/classnames-2.2.5.tgz"
-    },
-    "core-js": {
-      "version": "1.2.7",
-      "from": "core-js@^1.0.0",
-      "resolved": "https://registry.npmjs.org/core-js/-/core-js-1.2.7.tgz"
+      "version": "2.3.1",
+      "resolved": "https://registry.npmjs.org/classnames/-/classnames-2.3.1.tgz",
+      "integrity": "sha512-OlQdbZ7gLfGarSqxesMesDa5uz7KFbID8Kpq/SxIoNGDqY8lSYs0D+hhtBXhcdB3rcbXArFr7vlHheLk1voeNA=="
     },
-    "create-react-class": {
-      "version": "15.6.3",
-      "from": "create-react-class@^15.6.0",
-      "resolved": "https://registry.npmjs.org/create-react-class/-/create-react-class-15.6.3.tgz"
+    "direction": {
+      "version": "1.0.4",
+      "resolved": "https://registry.npmjs.org/direction/-/direction-1.0.4.tgz",
+      "integrity": "sha512-GYqKi1aH7PJXxdhTeZBFrg8vUBeKXi+cNprXsC1kpJcbcVnV9wBsrOu1cQEdG0WeQwlfHiy3XvnKfIrJ2R0NzQ=="
     },
     "dom-walk": {
-      "version": "0.1.1",
-      "from": "dom-walk@^0.1.0",
-      "resolved": "https://registry.npmjs.org/dom-walk/-/dom-walk-0.1.1.tgz"
-    },
-    "eases": {
-      "version": "1.0.8",
-      "from": "eases@~1.0.8",
-      "resolved": "https://registry.npmjs.org/eases/-/eases-1.0.8.tgz"
-    },
-    "encoding": {
-      "version": "0.1.12",
-      "from": "encoding@^0.1.11",
-      "resolved": "https://registry.npmjs.org/encoding/-/encoding-0.1.12.tgz"
-    },
-    "fbjs": {
-      "version": "0.8.16",
-      "from": "fbjs@^0.8.16",
-      "resolved": "https://registry.npmjs.org/fbjs/-/fbjs-0.8.16.tgz"
-    },
-    "for-each": {
-      "version": "0.3.2",
-      "from": "for-each@^0.3.2",
-      "resolved": "https://registry.npmjs.org/for-each/-/for-each-0.3.2.tgz"
+      "version": "0.1.2",
+      "resolved": "https://registry.npmjs.org/dom-walk/-/dom-walk-0.1.2.tgz",
+      "integrity": "sha512-6QvTW9mrGeIegrFXdtQi9pk7O/nSK6lSdXW2eqUspN5LWD7UTji2Fqw5V2YLjBpHEoU9Xl/eUWNpDeZvoyOv2w=="
     },
     "global": {
-      "version": "4.3.2",
-      "from": "global@~4.3.0",
-      "resolved": "https://registry.npmjs.org/global/-/global-4.3.2.tgz"
+      "version": "4.4.0",
+      "resolved": "https://registry.npmjs.org/global/-/global-4.4.0.tgz",
+      "integrity": "sha512-wv/LAoHdRE3BeTGz53FAamhGlPLhlssK45usmGFThIi4XqnBmjKQ16u+RNbP7WvigRZDxUsM0J3gcQ5yicaL0w==",
+      "requires": {
+        "min-document": "^2.19.0",
+        "process": "^0.11.10"
+      }
     },
     "hoist-non-react-statics": {
-      "version": "1.2.0",
-      "from": "hoist-non-react-statics@^1.0.0",
-      "resolved": "https://registry.npmjs.org/hoist-non-react-statics/-/hoist-non-react-statics-1.2.0.tgz"
+      "version": "3.3.2",
+      "resolved": "https://registry.npmjs.org/hoist-non-react-statics/-/hoist-non-react-statics-3.3.2.tgz",
+      "integrity": "sha512-/gGivxi8JPKWNm/W0jSmzcMPpfpPLc3dY/6GxhX2hQ9iGj3aDfklV4ET7NjKpSinLpJ5vafa9iiGIEZg10SfBw==",
+      "requires": {
+        "react-is": "^16.7.0"
+      },
+      "dependencies": {
+        "react-is": {
+          "version": "16.13.1",
+          "resolved": "https://registry.npmjs.org/react-is/-/react-is-16.13.1.tgz",
+          "integrity": "sha512-24e6ynE2H+OKt4kqsOvNd8kBpV65zoxbA4BVsEOB3ARVWQki/DHzaUoC5KuON/BiccDaCCTZBuOcfZs70kR8bQ=="
+        }
+      }
     },
-    "iconv-lite": {
-      "version": "0.4.19",
-      "from": "iconv-lite@~0.4.13",
-      "resolved": "https://registry.npmjs.org/iconv-lite/-/iconv-lite-0.4.19.tgz"
+    "ilib": {
+      "version": "14.14.0",
+      "resolved": "https://registry.npmjs.org/ilib/-/ilib-14.14.0.tgz",
+      "integrity": "sha512-QuhH6qwJYq8c368+px4KA3KXfWfRC8XDt/zGNdwOz5C19ZOkPaxIh9LRr1RmzHEPJsd0mKj/JGV3pnzx1xXljg=="
     },
     "invariant": {
       "version": "2.2.4",
-      "from": "invariant@~2.2.2",
-      "resolved": "https://registry.npmjs.org/invariant/-/invariant-2.2.4.tgz"
+      "resolved": "https://registry.npmjs.org/invariant/-/invariant-2.2.4.tgz",
+      "integrity": "sha512-phJfQVBuaJM5raOpJjSfkiD6BpbCE4Ns//LaXl6wGYtUBY83nWS6Rf9tXm2e8VaK60JEjYldbPif/A2B1C2gNA==",
+      "requires": {
+        "loose-envify": "^1.0.0"
+      }
     },
     "is-function": {
-      "version": "1.0.1",
-      "from": "is-function@^1.0.1",
-      "resolved": "https://registry.npmjs.org/is-function/-/is-function-1.0.1.tgz"
-    },
-    "is-stream": {
-      "version": "1.1.0",
-      "from": "is-stream@^1.0.1",
-      "resolved": "https://registry.npmjs.org/is-stream/-/is-stream-1.1.0.tgz"
-    },
-    "isomorphic-fetch": {
-      "version": "2.2.1",
-      "from": "isomorphic-fetch@^2.1.1",
-      "resolved": "https://registry.npmjs.org/isomorphic-fetch/-/isomorphic-fetch-2.2.1.tgz"
+      "version": "1.0.2",
+      "resolved": "https://registry.npmjs.org/is-function/-/is-function-1.0.2.tgz",
+      "integrity": "sha512-lw7DUp0aWXYg+CBCN+JKkcE0Q2RayZnSvnZBlwgxHBQhqt5pZNVy4Ri7H9GmmXkdu7LUthszM+Tor1u/2iBcpQ=="
     },
     "js-tokens": {
-      "version": "3.0.2",
-      "from": "js-tokens@^3.0.0",
-      "resolved": "https://registry.npmjs.org/js-tokens/-/js-tokens-3.0.2.tgz"
+      "version": "4.0.0",
+      "resolved": "https://registry.npmjs.org/js-tokens/-/js-tokens-4.0.0.tgz",
+      "integrity": "sha512-RdJUflcE3cUzKiMqQgsCu06FPu9UdIJO0beYbPhHN4k6apgJtifcoCtT9bcxOpYBtpD2kCM6Sbzg4CausW/PKQ=="
     },
     "lodash": {
-      "version": "4.17.5",
-      "from": "lodash@^4.17.5",
-      "resolved": "https://registry.npmjs.org/lodash/-/lodash-4.17.5.tgz"
+      "version": "4.17.21",
+      "resolved": "https://registry.npmjs.org/lodash/-/lodash-4.17.21.tgz",
+      "integrity": "sha512-v2kDEe57lecTulaDIuNTPy3Ry4gLGJ6Z1O3vE1krgXZNrsQ+LFTGHVxVjcXPs17LhbZVGedAJv8XZ1tvj5FvSg=="
     },
     "lodash-es": {
-      "version": "4.17.7",
-      "from": "lodash-es@^4.17.5",
-      "resolved": "https://registry.npmjs.org/lodash-es/-/lodash-es-4.17.7.tgz"
+      "version": "4.17.21",
+      "resolved": "https://registry.npmjs.org/lodash-es/-/lodash-es-4.17.21.tgz",
+      "integrity": "sha512-mKnC+QJ9pWVzv+C4/U3rRsHapFfHvQFoFB92e52xeyGMcX6/OlIl78je1u8vePzYZSkkogMPJ2yjxxsb89cxyw=="
     },
     "loose-envify": {
-      "version": "1.3.1",
-      "from": "loose-envify@^1.0.0",
-      "resolved": "https://registry.npmjs.org/loose-envify/-/loose-envify-1.3.1.tgz"
+      "version": "1.4.0",
+      "resolved": "https://registry.npmjs.org/loose-envify/-/loose-envify-1.4.0.tgz",
+      "integrity": "sha512-lyuxPGr/Wfhrlem2CL/UcnUc1zcqKAImBDzukY7Y5F/yQiNdko6+fRLevlw1HgMySw7f611UIY408EtxRSoK3Q==",
+      "requires": {
+        "js-tokens": "^3.0.0 || ^4.0.0"
+      }
     },
     "min-document": {
       "version": "2.19.0",
-      "from": "min-document@^2.19.0",
-      "resolved": "https://registry.npmjs.org/min-document/-/min-document-2.19.0.tgz"
-    },
-    "node-fetch": {
-      "version": "1.7.3",
-      "from": "node-fetch@^1.0.1",
-      "resolved": "https://registry.npmjs.org/node-fetch/-/node-fetch-1.7.3.tgz"
+      "resolved": "https://registry.npmjs.org/min-document/-/min-document-2.19.0.tgz",
+      "integrity": "sha512-9Wy1B3m3f66bPPmU5hdA4DR4PB2OfDU/+GS3yAB7IQozE3tqXaVv2zOjgla7MEGSRv95+ILmOuvhLkOK6wJtCQ==",
+      "requires": {
+        "dom-walk": "^0.1.0"
+      }
     },
     "object-assign": {
       "version": "4.1.1",
-      "from": "object-assign@^4.1.1",
-      "resolved": "https://registry.npmjs.org/object-assign/-/object-assign-4.1.1.tgz"
+      "resolved": "https://registry.npmjs.org/object-assign/-/object-assign-4.1.1.tgz",
+      "integrity": "sha512-rJgTQnkUnH1sFw8yT6VSU3zD3sWmu6sZhIseY8VX+GRu3P6F7Fu+JNDoXfklElbLJSnc3FUQHVe4cU5hj+BcUg=="
     },
     "parse-headers": {
-      "version": "2.0.1",
-      "from": "parse-headers@^2.0.0",
-      "resolved": "https://registry.npmjs.org/parse-headers/-/parse-headers-2.0.1.tgz"
+      "version": "2.0.5",
+      "resolved": "https://registry.npmjs.org/parse-headers/-/parse-headers-2.0.5.tgz",
+      "integrity": "sha512-ft3iAoLOB/MlwbNXgzy43SWGP6sQki2jQvAyBg/zDFAgr9bfNWZIUj42Kw2eJIl8kEi4PbgE6U1Zau/HwI75HA=="
     },
     "process": {
-      "version": "0.5.2",
-      "from": "process@~0.5.1",
-      "resolved": "https://registry.npmjs.org/process/-/process-0.5.2.tgz"
-    },
-    "promise": {
-      "version": "7.3.1",
-      "from": "promise@^7.1.1",
-      "resolved": "https://registry.npmjs.org/promise/-/promise-7.3.1.tgz"
+      "version": "0.11.10",
+      "resolved": "https://registry.npmjs.org/process/-/process-0.11.10.tgz",
+      "integrity": "sha512-cdGef/drWFoydD1JsMzuFf8100nZl+GT+yacc2bEced5f9Rjk4z+WtFUTBu9PhOi9j/jfmBPu0mMEY4wIdAF8A=="
     },
     "prop-types": {
-      "version": "15.6.1",
-      "from": "prop-types@^15.5.10",
-      "resolved": "https://registry.npmjs.org/prop-types/-/prop-types-15.6.1.tgz"
+      "version": "15.8.1",
+      "resolved": "https://registry.npmjs.org/prop-types/-/prop-types-15.8.1.tgz",
+      "integrity": "sha512-oj87CgZICdulUohogVAR7AjlC0327U4el4L6eAvOqCeudMDVU0NThNaV+b9Df4dXgSP1gXMTnPdhfe/2qDH5cg==",
+      "requires": {
+        "loose-envify": "^1.4.0",
+        "object-assign": "^4.1.1",
+        "react-is": "^16.13.1"
+      },
+      "dependencies": {
+        "react-is": {
+          "version": "16.13.1",
+          "resolved": "https://registry.npmjs.org/react-is/-/react-is-16.13.1.tgz",
+          "integrity": "sha512-24e6ynE2H+OKt4kqsOvNd8kBpV65zoxbA4BVsEOB3ARVWQki/DHzaUoC5KuON/BiccDaCCTZBuOcfZs70kR8bQ=="
+        }
+      }
     },
     "ramda": {
-      "version": "0.24.1",
-      "from": "ramda@~0.24.1",
-      "resolved": "https://registry.npmjs.org/ramda/-/ramda-0.24.1.tgz"
+      "version": "0.28.0",
+      "resolved": "https://registry.npmjs.org/ramda/-/ramda-0.28.0.tgz",
+      "integrity": "sha512-9QnLuG/kPVgWvMQ4aODhsBUFKOUmnbUnsSXACv+NCQZcHbeb+v8Lodp8OVxtRULN1/xOyYLLaL6npE6dMq5QTA=="
     },
     "react": {
-      "version": "15.6.2",
-      "from": "react@^15.6.1",
-      "resolved": "https://registry.npmjs.org/react/-/react-15.6.2.tgz"
+      "version": "18.2.0",
+      "resolved": "https://registry.npmjs.org/react/-/react-18.2.0.tgz",
+      "integrity": "sha512-/3IjMdb2L9QbBdWiW5e3P2/npwMBaU9mHCSCUzNln0ZCYbcfTsGbTJrU/kGemdH2IWmB2ioZ+zkxtmq6g09fGQ==",
+      "requires": {
+        "loose-envify": "^1.1.0"
+      }
     },
     "react-dom": {
-      "version": "15.6.2",
-      "from": "react-dom@^15.6.1",
-      "resolved": "https://registry.npmjs.org/react-dom/-/react-dom-15.6.2.tgz"
+      "version": "18.2.0",
+      "resolved": "https://registry.npmjs.org/react-dom/-/react-dom-18.2.0.tgz",
+      "integrity": "sha512-6IMTriUmvsjHUjNtEDudZfuDQUoWXVxKHhlEGSk81n4YFS+r/Kl99wXiwlVXtPBtJenozv2P+hxDsw9eA7Xo6g==",
+      "requires": {
+        "loose-envify": "^1.1.0",
+        "scheduler": "^0.23.0"
+      }
+    },
+    "react-is": {
+      "version": "18.2.0",
+      "resolved": "https://registry.npmjs.org/react-is/-/react-is-18.2.0.tgz",
+      "integrity": "sha512-xWGDIW6x921xtzPkhiULtthJHoJvBbF3q26fzloPCK0hsvxtPVelvftw3zjbHWSkR2km9Z+4uxbDDK/6Zw9B8w=="
+    },
+    "react-lifecycles-compat": {
+      "version": "3.0.4",
+      "resolved": "https://registry.npmjs.org/react-lifecycles-compat/-/react-lifecycles-compat-3.0.4.tgz",
+      "integrity": "sha512-fBASbA6LnOU9dOU2eW7aQ8xmYBSXUIWr+UmF9b1efZBazGNO+rcXT/icdKnYm2pTwcRylVUYwW7H1PHfLekVzA=="
     },
     "react-redux": {
-      "version": "5.0.7",
-      "from": "react-redux@^5.0.7",
-      "resolved": "https://registry.npmjs.org/react-redux/-/react-redux-5.0.7.tgz",
+      "version": "5.1.2",
+      "resolved": "https://registry.npmjs.org/react-redux/-/react-redux-5.1.2.tgz",
+      "integrity": "sha512-Ns1G0XXc8hDyH/OcBHOxNgQx9ayH3SPxBnFCOidGKSle8pKihysQw2rG/PmciUQRoclhVBO8HMhiRmGXnDja9Q==",
+      "requires": {
+        "@babel/runtime": "^7.1.2",
+        "hoist-non-react-statics": "^3.3.0",
+        "invariant": "^2.2.4",
+        "loose-envify": "^1.1.0",
+        "prop-types": "^15.6.1",
+        "react-is": "^16.6.0",
+        "react-lifecycles-compat": "^3.0.0"
+      },
       "dependencies": {
-        "hoist-non-react-statics": {
-          "version": "2.5.0",
-          "from": "hoist-non-react-statics@^2.5.0",
-          "resolved": "https://registry.npmjs.org/hoist-non-react-statics/-/hoist-non-react-statics-2.5.0.tgz"
+        "react-is": {
+          "version": "16.13.1",
+          "resolved": "https://registry.npmjs.org/react-is/-/react-is-16.13.1.tgz",
+          "integrity": "sha512-24e6ynE2H+OKt4kqsOvNd8kBpV65zoxbA4BVsEOB3ARVWQki/DHzaUoC5KuON/BiccDaCCTZBuOcfZs70kR8bQ=="
         }
       }
     },
-    "recompose": {
-      "version": "0.23.5",
-      "from": "recompose@~0.23.5",
-      "resolved": "https://registry.npmjs.org/recompose/-/recompose-0.23.5.tgz"
-    },
     "redux": {
       "version": "3.7.2",
-      "from": "redux@^3.7.2",
-      "resolved": "https://registry.npmjs.org/redux/-/redux-3.7.2.tgz"
+      "resolved": "https://registry.npmjs.org/redux/-/redux-3.7.2.tgz",
+      "integrity": "sha512-pNqnf9q1hI5HHZRBkj3bAngGZW/JMCmexDlOxw4XagXY2o1327nHH54LoTjiPJ0gizoqPDRqWyX/00g0hD6w+A==",
+      "requires": {
+        "lodash": "^4.2.1",
+        "lodash-es": "^4.2.1",
+        "loose-envify": "^1.1.0",
+        "symbol-observable": "^1.0.3"
+      }
     },
     "redux-thunk": {
-      "version": "2.2.0",
-      "from": "redux-thunk@^2.2.0",
-      "resolved": "https://registry.npmjs.org/redux-thunk/-/redux-thunk-2.2.0.tgz"
-    },
-    "setimmediate": {
-      "version": "1.0.5",
-      "from": "setimmediate@^1.0.5",
-      "resolved": "https://registry.npmjs.org/setimmediate/-/setimmediate-1.0.5.tgz"
+      "version": "2.4.1",
+      "resolved": "https://registry.npmjs.org/redux-thunk/-/redux-thunk-2.4.1.tgz",
+      "integrity": "sha512-OOYGNY5Jy2TWvTL1KgAlVy6dcx3siPJ1wTq741EPyUKfn6W6nChdICjZwCd0p8AZBs5kWpZlbkXW2nE/zjUa+Q==",
+      "requires": {}
+    },
+    "regenerator-runtime": {
+      "version": "0.13.9",
+      "resolved": "https://registry.npmjs.org/regenerator-runtime/-/regenerator-runtime-0.13.9.tgz",
+      "integrity": "sha512-p3VT+cOEgxFsRRA9X4lkI1E+k2/CtnKtU4gcxyaCUreilL/vqI6CdZ3wxVUx3UOUg+gnUOQQcRI7BmSI656MYA=="
+    },
+    "scheduler": {
+      "version": "0.23.0",
+      "resolved": "https://registry.npmjs.org/scheduler/-/scheduler-0.23.0.tgz",
+      "integrity": "sha512-CtuThmgHNg7zIZWAXi3AsyIzA3n4xx7aNyjwC2VJldO2LMVDhFK+63xGqq6CsJH4rTAt6/M+N4GhZiDYPx9eUw==",
+      "requires": {
+        "loose-envify": "^1.1.0"
+      }
     },
     "symbol-observable": {
       "version": "1.2.0",
-      "from": "symbol-observable@^1.0.4",
-      "resolved": "https://registry.npmjs.org/symbol-observable/-/symbol-observable-1.2.0.tgz"
-    },
-    "trim": {
-      "version": "0.0.1",
-      "from": "trim@0.0.1",
-      "resolved": "https://registry.npmjs.org/trim/-/trim-0.0.1.tgz"
-    },
-    "ua-parser-js": {
-      "version": "0.7.17",
-      "from": "ua-parser-js@^0.7.9",
-      "resolved": "https://registry.npmjs.org/ua-parser-js/-/ua-parser-js-0.7.17.tgz"
+      "resolved": "https://registry.npmjs.org/symbol-observable/-/symbol-observable-1.2.0.tgz",
+      "integrity": "sha512-e900nM8RRtGhlV36KGEU9k65K3mPb1WV70OdjfxlG2EAuM1noi/E/BaW/uMhL7bPEssK8QV57vN3esixjUvcXQ=="
     },
     "warning": {
-      "version": "3.0.0",
-      "from": "warning@~3.0.0",
-      "resolved": "https://registry.npmjs.org/warning/-/warning-3.0.0.tgz"
-    },
-    "whatwg-fetch": {
-      "version": "2.0.3",
-      "from": "whatwg-fetch@>=0.10.0",
-      "resolved": "https://registry.npmjs.org/whatwg-fetch/-/whatwg-fetch-2.0.3.tgz"
+      "version": "4.0.3",
+      "resolved": "https://registry.npmjs.org/warning/-/warning-4.0.3.tgz",
+      "integrity": "sha512-rpJyN222KWIvHJ/F53XSZv0Zl/accqHR8et1kpaMTD/fLCRxtV8iX8czMzY7sVZupTI3zcUTg8eycS2kNF9l6w==",
+      "requires": {
+        "loose-envify": "^1.0.0"
+      }
     },
     "xhr": {
-      "version": "2.4.1",
-      "from": "xhr@~2.4.0",
-      "resolved": "https://registry.npmjs.org/xhr/-/xhr-2.4.1.tgz"
+      "version": "2.6.0",
+      "resolved": "https://registry.npmjs.org/xhr/-/xhr-2.6.0.tgz",
+      "integrity": "sha512-/eCGLb5rxjx5e3mF1A7s+pLlR6CGyqWN91fv1JgER5mVWg1MZmlhBvy9kjcsOdRk8RrIujotWyJamfyrp+WIcA==",
+      "requires": {
+        "global": "~4.4.0",
+        "is-function": "^1.0.1",
+        "parse-headers": "^2.0.0",
+        "xtend": "^4.0.0"
+      }
     },
     "xtend": {
-      "version": "4.0.1",
-      "from": "xtend@^4.0.0",
-      "resolved": "https://registry.npmjs.org/xtend/-/xtend-4.0.1.tgz"
+      "version": "4.0.2",
+      "resolved": "https://registry.npmjs.org/xtend/-/xtend-4.0.2.tgz",
+      "integrity": "sha512-LKYU1iAXJXUgAXn9URjiu+MWhyUXHsvfp7mcuYm9dSUKK0/CjtrUwFAxD82/mCWbtLsGjFIad0wIsod4zrTAEQ=="
     }
   }
 }
diff --git a/resource/csdk/stack/samples/webos/com.example.app.iotivity/package.json b/resource/csdk/stack/samples/webos/com.example.app.iotivity/package.json
index c09a70a1f..43f37fcb1 100644
--- a/resource/csdk/stack/samples/webos/com.example.app.iotivity/package.json
+++ b/resource/csdk/stack/samples/webos/com.example.app.iotivity/package.json
@@ -29,15 +29,15 @@
     "extends": "enact"
   },
   "dependencies": {
-    "@enact/core": "^1.8.0",
-    "@enact/i18n": "^1.8.0",
-    "@enact/moonstone": "^1.8.0",
-    "@enact/spotlight": "^1.8.0",
-    "@enact/ui": "^1.8.0",
-    "@enact/webos": "^1.8.0",
+    "@enact/core": "^4.5.0",
+    "@enact/i18n": "^4.5.0",
+    "@enact/moonstone": "^4.5.0",
+    "@enact/spotlight": "^4.5.0",
+    "@enact/ui": "^4.5.0",
+    "@enact/webos": "^4.5.0",
     "prop-types": "^15.5.10",
-    "react": "^15.6.1",
-    "react-dom": "^15.6.1",
+    "react": "^18.0.0",
+    "react-dom": "^18.0.0",
     "react-redux": "^5.0.7",
     "redux": "^3.7.2",
     "redux-thunk": "^2.2.0"
diff --git a/resource/csdk/stack/samples/webos/com.example.app.iotivity/src/App/App.js b/resource/csdk/stack/samples/webos/com.example.app.iotivity/src/App/App.js
index 0c1a9bcf8..a8e96fdac 100644
--- a/resource/csdk/stack/samples/webos/com.example.app.iotivity/src/App/App.js
+++ b/resource/csdk/stack/samples/webos/com.example.app.iotivity/src/App/App.js
@@ -1,4 +1,3 @@
-import kind from '@enact/core/kind';
 import MoonstoneDecorator from '@enact/moonstone/MoonstoneDecorator';
 import Panels from '@enact/moonstone/Panels';
 import React from 'react';
diff --git a/resource/csdk/stack/samples/webos/com.example.app.iotivity/src/index.js b/resource/csdk/stack/samples/webos/com.example.app.iotivity/src/index.js
index 4538c89c4..5c0ebf0cd 100644
--- a/resource/csdk/stack/samples/webos/com.example.app.iotivity/src/index.js
+++ b/resource/csdk/stack/samples/webos/com.example.app.iotivity/src/index.js
@@ -1,10 +1,11 @@
+/* global ENACT_PACK_ISOMORPHIC */
+
 import React from 'react';
-import {render} from 'react-dom';
+import {createRoot, hydrateRoot} from 'react-dom/client';
 import {createStore, applyMiddleware, compose} from 'redux';
 import {Provider} from 'react-redux';
 import reducer from './reducer/Reducer';
 import thunkMiddleware from 'redux-thunk';
-
 import App from './App';
 
 const composeEnhancers = (typeof window === 'object' && window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__) || compose;
@@ -17,7 +18,6 @@ function configureStore (initialState) {
             applyMiddleware(thunkMiddleware)
         )
     );
-
     return ret;
 }
 
@@ -29,12 +29,13 @@ let appElement = (
     </Provider>
 );
 
-// In a browser environment, render instead of exporting
+// In a browser environment, render the app to the document.
 if (typeof window !== 'undefined') {
-    render(
-        appElement,
-        document.getElementById('root')
-    );
+    if (ENACT_PACK_ISOMORPHIC) {
+        hydrateRoot(document.getElementById('root'), appElement);
+    } else {
+        createRoot(document.getElementById('root')).render(appElement);
+    }
 }
 
-export default appElement;
\ No newline at end of file
+export default appElement;
diff --git a/resource/csdk/stack/samples/webos/com.example.app.iotivity/src/views/Client.js b/resource/csdk/stack/samples/webos/com.example.app.iotivity/src/views/Client.js
index 382ef649b..662e8b7cc 100644
--- a/resource/csdk/stack/samples/webos/com.example.app.iotivity/src/views/Client.js
+++ b/resource/csdk/stack/samples/webos/com.example.app.iotivity/src/views/Client.js
@@ -2,9 +2,9 @@ import React from 'react';
 import { connect } from 'react-redux';
 import PropTypes from 'prop-types';
 
-import { MarqueeText } from '@enact/moonstone/Marquee';
+import Marquee from '@enact/moonstone/Marquee';
 import Button from '@enact/moonstone/Button';
-import Divider from '@enact/moonstone/Divider';
+import Heading from '@enact/moonstone/Heading';
 import CheckboxItem from '@enact/moonstone/CheckboxItem';
 import BodyText from '@enact/moonstone/BodyText';
 import SpotlightContainerDecorator from '@enact/moonstone/Scroller';
@@ -54,7 +54,7 @@ class Client extends React.Component {
         }
         this.props.stopClient();
     }
-    componentWillReceiveProps(props) {
+    UNSAFE_componentWillReceiveProps(props) {
         console.log(props);
         if (props.discoveredResources && props.discoveredResources.length > 0) {
             discoveredItems = [];
@@ -268,7 +268,7 @@ class Client extends React.Component {
                 </div>
                 <div style={{ "paddingLeft": 20 + "px", "paddingTop": 9 + "px" }}>
                     <div style={{ "width": 600 + "px", "display": "inline-block", "verticalAlign": "middle" }}>
-                        <MarqueeText marqueeOn="render">Change mode</MarqueeText>
+                        <Marquee marqueeOn="render">Change mode</Marquee>
                     </div>
                     <Button small onClick={this.onRFOTMTapped}>RFOTM</Button>
                     <Button small onClick={this.onRFNOPTapped}>RFNOP</Button>
@@ -290,7 +290,7 @@ class Client extends React.Component {
                         </div>
                     </SpotlightContainerDecorator>
                 </Layout>
-                <Divider />
+                <Heading />
                 <Layout align="start">
                     <SpotlightContainerDecorator focusableScrollbar style={{ "height": 200 + "px" }} direction="both" horizontalScrollbar="auto" verticalScrollbar="auto">
                         {discoveredItemDatas ?
@@ -324,7 +324,7 @@ class Client extends React.Component {
                                 <Button style={{ lineHeight: 25 + "px" }} disabled={!this.state.resourceUri || !this.state.resourceKey || !this.state.resourceValue} small onClick={this.onPostResourceTapped}>send</Button>
                             </div> : null}
                             <div>
-                                <MarqueeText marqueeOn="render">return</MarqueeText>
+                                <Marquee marqueeOn="render">return</Marquee>
                                 <SpotlightContainerDecorator focusableScrollbar style={{ "height": 400 + "px" }} direction="both" horizontalScrollbar="auto" verticalScrollbar="auto">
                                     <BodyText>{(detailResourceInfo ? JSON.stringify(detailResourceInfo, null, 4) : "")}</BodyText>
                                 </SpotlightContainerDecorator>
@@ -370,4 +370,4 @@ let mapStateToProps = (state) => {
         clientStartable: state.clientStartable
     };
 };
-export default connect(mapStateToProps, mapDispatchToProps)(Client);
\ No newline at end of file
+export default connect(mapStateToProps, mapDispatchToProps)(Client);
diff --git a/resource/csdk/stack/samples/webos/com.example.app.iotivity/src/views/MainPanel.js b/resource/csdk/stack/samples/webos/com.example.app.iotivity/src/views/MainPanel.js
index d5c718623..0308f9fb4 100644
--- a/resource/csdk/stack/samples/webos/com.example.app.iotivity/src/views/MainPanel.js
+++ b/resource/csdk/stack/samples/webos/com.example.app.iotivity/src/views/MainPanel.js
@@ -1,8 +1,7 @@
-import React from 'react';
 import kind from '@enact/core/kind';
 
 import { Panel, Header } from '@enact/moonstone/Panels';
-import Divider from '@enact/moonstone/Divider';
+import Heading from '@enact/moonstone/Heading';
 
 import Server from './Server';
 import Client from './Client';
@@ -16,11 +15,11 @@ const MainPanel = kind({
             <Header title="iotivity sampler" type="compact" />
             <Layout align="start">
                 <Cell>
-                    <Divider casing="word">Server role</Divider>
+                    <Heading casing="word">Server role</Heading>
                     <Server />
                 </Cell>
                 <Cell>
-                    <Divider casing="word">Client role</Divider>
+                    <Heading casing="word">Client role</Heading>
                     <Client />
                 </Cell>
             </Layout>
@@ -28,4 +27,4 @@ const MainPanel = kind({
     )
 });
 
-export default MainPanel;
\ No newline at end of file
+export default MainPanel;
diff --git a/resource/csdk/stack/samples/webos/com.example.app.iotivity/src/views/Server.js b/resource/csdk/stack/samples/webos/com.example.app.iotivity/src/views/Server.js
index 60e17be6d..b8d128d03 100644
--- a/resource/csdk/stack/samples/webos/com.example.app.iotivity/src/views/Server.js
+++ b/resource/csdk/stack/samples/webos/com.example.app.iotivity/src/views/Server.js
@@ -2,9 +2,9 @@ import React from 'react';
 import { connect } from 'react-redux';
 import PropTypes from 'prop-types';
 
-import { MarqueeText } from '@enact/moonstone/Marquee';
+import Marquee from '@enact/moonstone/Marquee';
 import Button from '@enact/moonstone/Button';
-import Divider from '@enact/moonstone/Divider';
+import Heading from '@enact/moonstone/Heading';
 
 import * as ActionCreators from '../actions/ActionCreators';
 
@@ -31,7 +31,7 @@ class Server extends React.Component {
         this.props.stopObserveBinarySwitchValue();
         this.props.stopServer();
     }
-    componentWillReceiveProps(props) {
+    UNSAFE_componentWillReceiveProps(props) {
         if (props.serverEnabled && !this.state.serverStarted)
         {
             this.setState({
@@ -104,7 +104,7 @@ class Server extends React.Component {
                     <Cell>
                         {serverEnabled ?
                             <div style={{ "lineHeight": 60 + "px" }}>
-                                <MarqueeText marqueeOn="render">Server is running</MarqueeText>
+                                <Marquee marqueeOn="render">Server is running</Marquee>
                             </div>
                             : null
                         }
@@ -112,17 +112,17 @@ class Server extends React.Component {
                 </Layout>
                 <Layout align="start">
                     <Cell>
-                        <Divider />
+                        <Heading />
                         {(serverEnabled && enableServerResourceControlUI) ?
                             <div>
                                 <div style={{ "paddingLeft": 20 + "px" }}>
-                                    <MarqueeText marqueeOn="render">Resource: binarySwitch</MarqueeText>
-                                    <MarqueeText marqueeOn="render">Type: oic.r.switch.binary</MarqueeText>
-                                    <MarqueeText marqueeOn="render">Observable: true</MarqueeText>
+                                    <Marquee marqueeOn="render">Resource: binarySwitch</Marquee>
+                                    <Marquee marqueeOn="render">Type: oic.r.switch.binary</Marquee>
+                                    <Marquee marqueeOn="render">Observable: true</Marquee>
                                 </div>
                                 <div style={{ "paddingLeft": 20 + "px", "paddingTop": 9 + "px" }}>
                                     <div style={{ "width": 600 + "px", "display": "inline-block", "verticalAlign": "middle" }}>
-                                        <MarqueeText marqueeOn="render">value</MarqueeText>
+                                        <Marquee marqueeOn="render">value</Marquee>
                                     </div>
                                     <Button small onClick={this.onResourceValueTapped}>{resourceValue ? "true" : "false"}</Button>
                                 </div>
-- 
2.17.1
